iter/filter_test.go

```diff
@@ -58,6 +58,17 @@ func TestFilter(t *testing.T) {
 	assert.Equal(t, evens.Next().Unwrap(), 2)
 }
 
+func TestFilter_method(t *testing.T) {
+	evens := iter.Count().Filter(filters.IsEven[int])
+	assert.Equal(t, evens.Next().Unwrap(), 0)
+	assert.Equal(t, evens.Next().Unwrap(), 2)
+}
+
+func TestFilterFilter(t *testing.T) {
+	evens := iter.Count().Filter(filters.IsEven[int]).Filter(filters.GreaterThan[int](1))
+	assert.Equal(t, evens.Next().Unwrap(), 2)
+}
+
```

Comment:
Did you mean to create a `ExampleFilter_method` test instead? Examples are used in documentation. Take a look at the other example methods (for `ForEach`) to see how they work.